Automattic/woocommerce-payments
# Set UPE order payment method title before email is sent

Fixes #3841 

#### Changes proposed in this Pull Request

We're currently [filtering the payment gateway title](https://github.com/Automattic/woocommerce-payments/blob/d4983d2f9a587ecbabaea2b1a12547ded1d3bf07/includes/payment-methods/class-upe-payment-gateway.php#L107) to change the heading displayed on checkout to Popular Payment Methods. We then set the correct payment method title on the order [in the `process_payment` function](https://github.com/Automattic/woocommerce-payments/blob/d4983d2f9a587ecbabaea2b1a12547ded1d3bf07/includes/payment-methods/class-upe-payment-gateway.php#L579). The problem is that emails are already sent before this function fires and the payment method title is still Popular Payment Methods at that time. This PR hooks into the email, setting the correct payment method title, which ensures that emails arrive with the correct payment method title. 

[This adjustment to the CC payment methods `get_title()` method](https://github.com/Automattic/woocommerce-payments/blob/d4983d2f9a587ecbabaea2b1a12547ded1d3bf07/includes/payment-methods/class-cc-payment-method.php#L53) was necessary because this function expects `payment_details` to come in the shape of a charge from the payment intent which contains a top level `network` key. When getting the payment method details via the api client, the network is nested inside a `networks` array. 









#### Testing instructions



1. Enable UPE checkout (legacy) using WCPay Dev Tools
2. Add a product to the cart and proceed to checkout
3. Ensure that Popular Payments Methods is the heading for WooCommerce Payments payment methods
4. Complete checkout using credit card
5. Check for the correct payment method title in the emails sent, such as the New Order and Processing Order emails. The [WP Mail Logging](https://wordpress.org/plugins/wp-mail-logging/) is helpful for catching emails. For example, *Visa credit card* or similar. 

-------------------


**Post merge**
